### PR TITLE
[ONNX] Add module name as PythonOp attribute

### PR DESCRIPTION
**Pull request by shubhambhokare1**
Stores the module name of the autograd function as attribute for a PythonOp node in the IR graph.

Before this modification:
```
graph(%input : Float(1, strides=[1], requires_grad=0, device=cpu)):
  %3 : Float(1, strides=[1], requires_grad=0, device=cpu) = ^MyExp[inplace=0]()(%input)]
  %6 : Float(1, strides=[1], requires_grad=0, device=cpu) = ^MyExp[inplace=0]()(%input)]
  %7 : int = prim::Constant[value=1]()
  %8 : Float(1, strides=[1], requires_grad=0, device=cpu) = aten::add(%3, %6, %7)
  return (%8)
```
After this modification:
```
graph(%input : Float(1, strides=[1], requires_grad=0, device=cpu)):
  %3 : Float(1, strides=[1], requires_grad=0, device=cpu) = ^MyExp[module="autograd1", inplace=0]()(%input)]
  %6 : Float(1, strides=[1], requires_grad=0, device=cpu) = ^MyExp[module="autograd2", inplace=0]()(%input)]
  %7 : int = prim::Constant[value=1]()
  %8 : Float(1, strides=[1], requires_grad=0, device=cpu) = aten::add(%3, %6, %7)
  return (%8)
```

### CODE REVIEW

**Review by garymm — CHANGES_REQUESTED** [review]
1. Could you please add a test that demonstrates the new behavior?
2. The PR title is misleading. This change does not actually add the fully qualified name as an attribute, it adds the module name. I think it'd be perfectly reasonable to add the qualified name instead if that's what you want to do. Just use `__qualname__` instead of `__module__`. That may actually be better since it seems from [the docs](https://docs.python.org/3/reference/datamodel.html) that `__module__` may be `None` but `__qualname__` should always be set. But either way seems OK to me, just make sure the PR title matches the implementation.

File: torch/csrc/autograd/python_function.cpp [hunk 1]
```diff
@@ -559,6 +559,16 @@ static void _trace_post_record(
   }
 
   node->i_(jit::attr::inplace, is_inplace);
+  if (node) {
+    auto module_name = PyDict_GetItemString(((PyTypeObject*)op_obj)->tp_dict, "__module__");
+    if (module_name) {
+      const char *ptr = PyUnicode_AsUTF8(module_name);
+      if (ptr) {
+          auto modname = std::string(ptr);
+          node->s_(jit::attr::module, modname);
```
Comment (garymm) [inline torch/csrc/autograd/python_function.cpp:568]: Suggestion for slightly more compact code without loss of clarity:
```suggestion
          node->s_(jit::attr::module, std::string(ptr));
```

**Review by garymm — APPROVED** [review]
Thanks!

File: torch/csrc/autograd/python_function.cpp [hunk 2]
```diff
@@ -559,6 +559,16 @@ static void _trace_post_record(
   }
 
   node->i_(jit::attr::inplace, is_inplace);
+  if (node) {
+    auto module_name = PyDict_GetItemString(((PyTypeObject*)op_obj)->tp_dict, "__module__");
+    if (module_name) {
```
Comment (BowenBao) [inline torch/csrc/autograd/python_function.cpp:564]: nit: more compat code, same with `ptr` below.
```suggestion
    if (auto module_name = PyDict_GetItemString(((PyTypeObject*)op_obj)->tp_dict, "__module__")) {
```

**Review by BowenBao — APPROVED** [review]
LGTM
